src/debugger/scriptImporter.ts

```diff
@@ -86,4 +87,14 @@ export class ScriptImporter {
                 return new FileSystem().writeFile(sourceMappingLocalPath, updatedContent);
             });
     }
+
+    /**
+     * Changes the port of the url to be the one configured as this.packagerPort
+     */
+    private overridePackagerPort(urlToOverride: string): string {
```

Comment:
I think this solution is good enough for now. If the team feels strongly about changing this to a better design I can spend some time doing that, if not I'd leave it as-is for the time being.